### PR TITLE
LOG-3569: Add functionality to enable TLS Security feature gate

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Adds functionality to enable a preview feature gate for TLS Security Profile
* Adds an option for consumption by the collector config generator

### Links
 * https://issues.redhat.com/browse/LOG-3569

cc @syedriko

### CODE REVIEW

File: internal/k8shandler/forwarding_features_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		} else {
+			Expect(options[enabledOption]).To(Equal(value), "Exp the option to equal the given value")
+		}
+
+	},
+		Entry("enables TLS Security profile for enabled", constants.PreviewTLSSecurityProfile, "", constants.PreviewTLSSecurityProfile, "enabled"),
+		Entry("enables TLS Security profile for eEabled", constants.PreviewTLSSecurityProfile, "", constants.PreviewTLSSecurityProfile, "eNabled"),
+		Entry("disables TLS Security profile for true", "", "", constants.PreviewTLSSecurityProfile, "true"),
+		Entry("enables old remote syslog for enabled", UseOldRemoteSyslogPlugin, "", UseOldRemoteSyslogPlugin, "enabled"),
+		Entry("enables old remote syslog for eEabled", UseOldRemoteSyslogPlugin, "", UseOldRemoteSyslogPlugin, "eNabled"),
```
Comment (syedriko) [inline internal/k8shandler/forwarding_features_test.go:49]: @jcantrill eEabled sp. here and a few lines above

File: internal/k8shandler/forwarding.go [hunk 2]
```diff
@@ -12,17 +13,29 @@ import (
...
-	if clusterRequest.ForwarderRequest == nil {
-		return false
+// EvaluateAnnotationsForEnabledCapabilities populates generator options with capabilities enabled by the ClusterLogForwarder
+func EvaluateAnnotationsForEnabledCapabilities(forwarder *logging.ClusterLogForwarder, options generator.Options) {
+	if forwarder == nil {
+		return
+	}
+	for key, value := range forwarder.Annotations {
+		switch key {
+		case constants.PreviewTLSSecurityProfile:
```
Comment (vparfonov) [inline internal/k8shandler/forwarding.go:27]: Do nothing here, maybe just remove this `case`

File: internal/generator/options.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+// Options is a map of Options used to customize the config generation. E.g. Debugging, legacy config generation
+type Options map[string]interface{}
+
+// NoOptions is used to pass empty options
+var NoOptions = Options{}
+
+// Has takes a key and returns true if it exists
+func (o Options) Has(key string) bool {
+	if _, found := o[key]; found {
```
Comment (syedriko) [inline internal/generator/options.go:11]:
```
_, found := o[key]
return found
```

File: internal/generator/options_test.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	. "github.com/onsi/gomega"
+	. "github.com/openshift/cluster-logging-operator/internal/generator"
+)
+
+var _ = Describe("Options#Has", func() {
+
+	var (
+		options = Options{}
+	)
+	It("should should be false when the key does not exist", func() {
```
Comment (syedriko) [inline internal/generator/options_test.go:14]: ~~should~~

File: internal/generator/options_test.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+		Expect(options.Has("foo")).To(BeFalse())
+	})
+
+	It("should should be true when the key exists", func() {
```
Comment (syedriko) [inline internal/generator/options_test.go:18]: ~~should~~

File: internal/k8shandler/forwarding.go (hunk 2 above)
Comment (syedriko) [reply internal/k8shandler/forwarding.go:27]: It is doing something for constants.PreviewTLSSecurityProfile, which is the same thing as for constants.UseOldRemoteSyslogPlugin. This is the same as
```
switch key {
case constants.PreviewTLSSecurityProfile, constants.UseOldRemoteSyslogPlugin:
...
```

File: internal/k8shandler/forwarding_features_test.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+var _ = Describe("#EvaluateAnnotationsForEnabledCapabilities", func() {
+
+	It("should do nothing if the forwarder is nil", func() {
+		options := generator.Options{}
+		EvaluateAnnotationsForEnabledCapabilities(nil, options)
+		Expect(options).To(BeEmpty(), "Exp no entries added to the options")
+	})
+	DescribeTable("when forwarder is not nil", func(enabledOption, value string, annotations ...string) {
+		pairs := annotations
+		if len(pairs)%2 != 0 {
```
Comment (syedriko) [inline internal/k8shandler/forwarding_features_test.go:23]: Just name `annotations` `pairs` and drop
```
pairs := annotations
```
?

File: internal/k8shandler/forwarding.go [hunk 7]
```diff
@@ -12,17 +13,29 @@ import (
...
+
 	corev1 "k8s.io/api/core/v1"
 )
 
-// useOldRemoteSyslogPlugin checks if old plugin (docebo/fluent-plugin-remote-syslog) is to be used for sending syslog or new plugin (dlackty/fluent-plugin-remote_syslog) is to be used
-func (clusterRequest *ClusterLoggingRequest) useOldRemoteSyslogPlugin() bool {
-	if clusterRequest.ForwarderRequest == nil {
-		return false
+// EvaluateAnnotationsForEnabledCapabilities populates generator options with capabilities enabled by the ClusterLogForwarder
+func EvaluateAnnotationsForEnabledCapabilities(forwarder *logging.ClusterLogForwarder, options generator.Options) {
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply internal/k8shandler/forwarding.go:21]: Updated test to require it to be exported :wink:

**Pull request merged**